packages/vm/src/instance.rs

```diff
@@ -121,7 +121,14 @@ where
         gas_limit: u64,
     ) -> Self {
         set_gas(&mut wasmer_instance, gas_limit);
-        let instance_ptr = NonNull::from(&mut wasmer_instance);
+
+        // The pointer shenanigans below are sound because:
```

Comment:
Seems like a safe way to get a reference, but not tested with actually calling it.

Reply:
true, still requires work, of course